File: 백준/Gold/23559. 밥/밥.py
n, x = map(int, input().split())
bob = sorted([list(map(int, input().split())) for _ in range(n)], key=lambda b: (b[0] - b[1]), reverse=True)
x -= 1000 * n # 천 원짜리 전부 구매
answer = sum([i[1] for i in bob]) # 천 원짜리 합
for b in bob:
    if x >= 4000 and b[0] > b[1]:
        answer += b[0] - b[1]
        x -= 4000
    else:
        break
print(answer)

File: 백준/Gold/23559. 밥/밥.js
const readline=require('readline')
const rl=readline.createInterface({
  input:process.stdin,
  output:process.stdout
})

// 23559

const input=[]
let iter=0
rl.on('line',(line)=>{
  input.push(line)
}).on('close',()=>{
  const [N,X]=input[iter++].split(' ').map(item=>Number(item))
  const l=[]
  let answer=0
  const diff=[]
  for(let i=0;i<N;i++){
    const [A,B]=input[iter++].split(' ').map(item=>Number(item))
    l.push([A,B])
    answer+=B
    if (A-B>0) diff.push(A-B)
  }

  let money=1000*N
  diff.sort((a,b)=>b-a)
  for(let i=0;i<diff.length;i++){
    if (money+4000>X) break
    answer+=diff[i]
    money+=4000
  }

  console.log(answer)
  process.exit()
})
